test/Web3Spec/Encoding/Simple.purs

```diff
@@ -187,3 +195,22 @@ intNTests =
       it "can fail to encode larger negative int248" do
          let mgiven =  (uIntNFromBigNumber $ negate $ (embed $ 2) `pow` 255 +< 1) :: Maybe (UIntN (D2 :& D4 :& D8))
          mgiven `shouldEqual` Nothing
+
+
+falseOrObjectTests :: forall r. Spec r Unit
+falseOrObjectTests = 
+  describe "FalseOrObject tests" do
+    let opts = defaultOptions { unwrapSingleConstructors = true }
+
+    it "can decode FalseOrObject instances that are false" do
+      let decodedFalse = (runExcept $ decodeJSON "false") :: (Either (NonEmptyList ForeignError) (FalseOrObject SyncStatus))
+      decodedFalse `shouldEqual` (Right $ FalseOrObject Nothing)
+
+    it "can decode FalseOrObject instances that are objects" do
+      let decodedObj = runExceptT $ decodeJSON "{ \"startingBlock\": 0, \"currentBlock\": 1, \"highestBlock\": 2 }"
+      decodedObj `shouldEqual` (Identity $ Right $ FalseOrObject $ Just $ SyncStatus {startingBlock: intToBN 0, currentBlock: intToBN 1, highestBlock: intToBN 2})
+
+  where 
+    intToBN i = unsafePartial $ fromJust $ parseBigNumber decimal $ toStringAs decimal i
```

Comment:
instead of `intToBN` you can use our `Algebra Int BigNumber` instance method `embed` which here specializes to `embed :: Int -> BigNumber` . Check the BigNumber module for details on what other classes BigNumber implements, as well as ways to multiply, add other `Int`s, etc to it.

Reply:
Ahh cool, thanks. I figured there was some Int / BigNumber magic going on there but didn't look into it enough.